examples/mysql/README.md

```diff
@@ -0,0 +1,86 @@
+MySQL filter example
+===================
+
+In this example, we show how a [MySQL
+filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/network_filters/mysql_proxy_filter)
+can be used with the Envoy proxy. The Envoy proxy [configuration](./envoy.yaml)
+includes a MySQL filter that routes egress requests to a MySQL server instance
+running in a separate container.
+
+## Usage
+
+1. `docker-compose build`
+2. `docker-compose up`
+3. Issue MySQL commands using a client (such as `mysql`)
+
+## Sample Output
+
+Use `mysql` to issue some commands and verify they are routed via Envoy:
+
+```
+$ mysql -h 0.0.0.0 -P 1999 -u root --ssl-mode DISABLED
+... snip ...
+
+mysql> CREATE DATABASE test;
+Query OK, 1 row affected (0.00 sec)
+
+mysql> USE test;
+Database changed
+
+mysql> CREATE TABLE test ( text VARCHAR(255) );
+Query OK, 0 rows affected (0.02 sec)
+
+mysql> SELECT COUNT(*) FROM test;
++----------+
+| COUNT(*) |
++----------+
+|        0 |
++----------+
+1 row in set (0.00 sec)
+
+mysql> INSERT INTO test VALUES ('hello, world!');
+Query OK, 1 row affected (0.01 sec)
+
+mysql> SELECT COUNT(*) FROM test;
++----------+
+| COUNT(*) |
++----------+
+|        1 |
++----------+
+1 row in set (0.00 sec)
+
+mysql> exit
+Bye
+```
+
+Check that the egress stats were updated:
+
+```
+$ curl -s http://localhost:8001/stats?filter=egress_mysql
+mysql.egress_mysql.auth_switch_request: 1
+mysql.egress_mysql.decoder_errors: 0
+mysql.egress_mysql.login_attempts: 1
+mysql.egress_mysql.login_failures: 0
+mysql.egress_mysql.protocol_errors: 21
+mysql.egress_mysql.queries_parse_error: 1
+mysql.egress_mysql.queries_parsed: 4
+mysql.egress_mysql.sessions: 1
+mysql.egress_mysql.upgraded_to_ssl: 0
+```
```

Comment:
It seems like there are a lot of protocol errors. Can you try using `mysql:5.5` for the Docker image and try this example again?

Reply:
I ran this with again with `5.5.62` and I still see protocol errors. This might be due to me using the `8.0.13` client on macOS. Running a 5.5 client from inside another container gets rid of the protocol errors. I'll update.

Reply:
Awesome, thanks.